packages/react-dom/src/events/DOMEventResponderSystem.js

```diff
@@ -158,16 +158,21 @@ const eventResponderContext: ReactDOMResponderContext = {
     validateResponderContext();
     const childFiber = getClosestInstanceFromNode(childTarget);
     const parentFiber = getClosestInstanceFromNode(parentTarget);
-    const parentAlternateFiber = parentFiber.alternate;
 
-    let node = childFiber;
-    while (node !== null) {
-      if (node === parentFiber || node === parentAlternateFiber) {
-        return true;
+    if (childFiber != null && parentFiber != null) {
+      const parentAlternateFiber = parentFiber.alternate;
+      let node = childFiber;
+      while (node !== null) {
+        if (node === parentFiber || node === parentAlternateFiber) {
+          return true;
+        }
+        node = node.return;
       }
-      node = node.return;
+      return false;
+    } else {
+      // Fallback to DOM APIs
```

Comment:
...and you might end up with an unmanaged DOM node here is when the target is not managed by React, e.g., it's the document or any other DOM context that React is rendering into. At the moment some of the responders cancel the gesture if a root level scroll event came from an ancestral element.